include/util/static_graph.hpp

```diff
@@ -148,19 +148,32 @@ template <typename EdgeDataT, bool UseSharedMemory = false> class StaticGraph
         return SPECIAL_EDGEID;
     }
 
-    // searches for a specific edge
-    EdgeIterator FindSmallestEdge(const NodeIterator from, const NodeIterator to) const
+    /**
+     * Finds the edge with the smallest `.distance` going from `from` to `to`
+     * @param from the source node ID
+     * @param to the target node ID
+     * @param filter a functor that returns a `bool` that determines whether an edge should be
+     * tested or not.
+     *   Takes `EdgeData` as a parameter.
+     * @return the ID of the smallest edge if any were found that satisfied *filter*, or
+     * `SPECIAL_EDGEID` if no
+     *   matching edge is found.
+     */
+    template <typename FilterFunction>
+    EdgeIterator FindSmallestEdge(const NodeIterator from,
+                                  const NodeIterator to,
+                                  const FilterFunction filter) const
```

Comment:
Taking filter by const-value does not make sense; see comment above: either by && and forwarding or by value (since the functor could be stateful).